### DISCUSSION TITLE
Authenticating another application after next-auth authentication

### DISCUSSION THREAD

**Discussion by emsesc** [nextauthjs/next-auth#1274] [Help] [answered]
After users sign in on my application **for the first time**, I need it to automatically redirect to another page (Github App installation).  I am not using a database, so I can't use "newUser" on [pages](https://next-auth.js.org/configuration/pages), and the redirect [callback](https://next-auth.js.org/configuration/callbacks) isn't particularly helpful since it would redirect to the page every time the user wants to sign in or sign out.

New user signs in --> next-auth authenticates user --> redirects to another page to install Github App

What is the best way to implement this with next-auth?

**Comment by balazsorban44** [comment, marked as answer]
you can use a client-side redirect with Next.js's `useRouter().push` that you call after you got a valid `next-auth` user session.
